### DISCUSSION TITLE
different trajectory on the Staubl robot

### DISCUSSION THREAD

**Discussion by eduardsik** [visose/Robots#204] [Q&A] [answered]
While testing the Staubli robot, I experience the path deviation shown in the images below, and I cannot find the cause of the error.
The robot executes the program, which in this case is a curve traversal, but about 500 mm higher than what is defined in the simulation. I thought the tcp was set incorrectly, but the amount of sliding upwards is the same in all positions, even if I set the robot plane to negative and place the robot -200mm lower in the simulation.
Another interesting phenomenon is when the robot approaches the track, it always goes down about as much as the difference in height, then goes up and starts to announce the curve. 
Where should I look for the error, what am I not noticing?

The expected curve trajectory:
![Screenshot 2022-12-15 140550](https://user-images.githubusercontent.com/4760893/207866507-8c2dc592-4300-4bc8-ae9f-8186990dc2da.png)

The actual trajectory on the real robot:
![Screenshot 2022-12-15 140458](https://user-images.githubusercontent.com/4760893/207866659-66d51e28-bcef-4568-8f23-bdeec24c4571.png)

**Comment by visose** [comment]
Ah, does it match exactly the height of the first axis?

**Reply by visose** [reply]
>Another interesting phenomenon is when the robot approaches the track, it always goes down about as much as the difference in height, then goes up and starts to announce the curve.

This is likely due one being a joint motion and the rest cartesian target which will have this incorrect offset.

**Reply by eduardsik** [reply]
> Ah, does it match exactly the height of the first axis?

Yes, exacly 375mm.

**Reply by eduardsik** [reply]
> > Another interesting phenomenon is when the robot approaches the track, it always goes down about as much as the difference in height, then goes up and starts to announce the curve.
> 
> This is likely due one being a joint motion and the rest cartesian target which will have this incorrect offset.

The robot approaches the starting pont of the line (picture) and goes up 375mm and tarting the trajectory. The new path is on baseline. 

![Screenshot 2022-12-15 185007](https://user-images.githubusercontent.com/4760893/207932133-c1ba09ce-dd12-48e8-a237-dbde973d0491.png)
The robot goes here first, than 375mm up (as high as the first  joint) ans tarting the track. 



![Screenshot 2022-12-15 184932](https://user-images.githubusercontent.com/4760893/207932145-e57cc51f-feb1-40be-8e83-a47074f05167.png)
The new path is on baseline.

**Comment by visose** [comment]
Thank's I've flaged it as a bug, I'll check it out soon.

**Reply by eduardsik** [reply]
Thank you so mutch!

**Comment by vinceHuyghe** [comment, marked as answer]
Hi All,

This is due to the base plane of the staubli robots in the Robots plugin
not being at the base of the robot but at the the height of the first axis.
Although it works correctly for consistency with the way other brands work
in the plugin it would be good to fix it.

Best,
Vince

**Comment by eduardsik** [comment]
Hello Vince,

thank you for the fast bugfix. Now the approaching is good, the robot goes direcly to the starting point of the curve, on the good height. But when I run the code, gives me this error the controller: "VAL3 runtime error. Invalid movement command: invalid destination point or mdesc... (Program name)"
I also run the same program with v1.6.1 and runs without error, but on the wrong height. Only the v1.6.3 gives me this error. 
I tried to get closer to the problem, but I did not find any soluion. The program in GH gives me just warnings, but same in both versions (1.6.1 and 1.6.3)

Any idea how to fix it? Thanks, Eduard

**Reply by visose** [reply]
Hi @eduardsik, to not lose track, can you post this in the issue here: https://github.com/visose/Robots/issues/205

**Comment by PhantomStaubli** [comment]
Hi @eduardsik, Can you share a GH file on how you connect to staubli? or at least the program send component of it? I have read the manual in and out but got stuck and cannot seem to be able to connect to cs8 whatsoever. thanks very much. your help is much appreciated.

**Reply by eduardsik** [reply]
Hi, I'm just using an ftp folder and save module on grasshopper plugin. If you have different methods, I'm curious. What staubli did you have?

**Comment by PhantomStaubli** [comment]
Hi @visose can you please help with the above request? I am stuck with the plugin for staubli. Thank you!

**Reply by visose** [reply]
Hi, when I tested the plugin with live Staubli robots I just used the `Save Program` component and copied the files to the controller using a USB stick.

**Comment by PhantomStaubli** [comment]
@eduardsik @visose How would you go about online programing if you need to read sensory inputs from grippers and write it back to GH?

**Comment by eduardsik** [comment]
Unfortunately I do not have. But try to download from robodk, they have
everything.

*Eduard Sik*
Creative Engineer <http://about.me/eduardsik> || eduardsik.com ||
***@***.*** || +36 70 237 4151 || book me
<https://doodle.com/eduardsik>

**Reply by PhantomStaubli** [reply]
I did last week but the format was weird. How do you upload files to CS8? Can you share one example please? My controller does not read .pgx format when I save them and the save component of the plugin does not work either, so had to create my own. Do I need to create an entire folder structure and if so, how does the plugin help generate it? I appreciate if you could explain by example. Thanks

**Comment by PhantomStaubli** [comment]


**Reply by visose** [reply]
Hi @PhantomStaubli, for Staubli specific questions that are unrelated to this plugin you will have more luck asking on a Staubli robots forum.

**Reply by PhantomStaubli** [reply]
Ok that's fine. I still need to get the answers for any previous question regarding the plugin please.

**Reply by visose** [reply]
I think the questions above are answered, but would be better if you start one separate thread per question.

**Reply by PhantomStaubli** [reply]
OK, I created new thread.
